core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala

```diff
@@ -172,6 +172,8 @@ private[ui] class AllJobsPage(parent: JobsTab) extends WebUIPage("") {
 
     val jobEventJsonAsStrSeq = makeJobEvent(jobs)
     val executorEventJsonAsStrSeq = makeExecutorEvent(executors)
+    val offset = TimeZone.getTimeZone(System.getProperty("user.timezone"))
```

Comment:
I think this is what `TimeZone.getDefault` does, and it takes care of a few other details